### DISCUSSION TITLE
Startup of Viewer-Manager

### DISCUSSION THREAD

**Discussion by Paul8043** [bernhard-42/vscode-ocp-cad-viewer#25] [Q&A]
This is a correction of the behavior I have reported at the very end of issue #21.

> Quickstart build123d
This works only, if an editor is open. This is easily forgotten. Is there a chance to get rid of this restriction? Would it be possible to store this information in the viewer preferences? With the hope that this needs to be done only once. The Quickstart takes long and it seems that the startup-times vary strongly. I have started VS-Code with the correct mamba environment, but I see a lot of command-windows popping up which try to activate the same environment again, by using conda instead of mamba. Could not find out from where this behavior comes from. Could be that this has nothing to do with OCP-CAD-Viewer. But this generates very long startup-times (sometimes up to 20 secs).
>

My description was not completely correct. I have only used the  `Quickstart Build123d` once. The normal start I do with the green Open-Viewer-Button. The rest of the description is correct. If you have a lot of open editors, then a lot of windows-cmd.exe are poppoing up (They try to activate with conda my bd-env again. Do not know why this happens or whether this is really needed). But anyhow  the focus seems to be influenced by these popping windows. I have to re-click on an editor (second time), so that the ocp-cad-viewer could start.

I would appreciate very much, if the restriction of having an activated editor could be removed.

**Comment by bernhard-42** [comment]
> then a lot of windows-cmd.exe are poppoing up

Don't see it on my machine under Windows. Tried a few things including installations, but no cmd.exe appears. Could that come from another extension or a setting? I have no idea, Windows is best effort - we will never get friends. 
I don't really work or test under Windows (except there is a new feature I expect to be different on Windows)

> I would appreciate very much, if the restriction of having an activated editor could be removed.

Unfortunately, not possible in the near future. Getting the right Python interpreter is anyhow a mess and by having a Pyhton file open I force the user to select the right environment.

**Reply by Paul8043** [reply]
Could it not say  better: "Windows and me, we will never get friends". Windows installations are high-risk enterprises. Unfortuneately I have still a few applications that run only under Windows, otherwise I would immediately switch to Linux. I will move my CAD-applications, when UBUNTU 24.04 is out.

> Unfortunately, not possible in the near future.

`Getting the right Python interpreter` is really complicated stuff. Never know how many python-interpreters are running somewhere on my machine. And VS-Code has so many possibilities to define settings. Looks much like a dense virgin forest.

Thought that when I create a dedicated environment for Build123d and if this environment gets activated before I start "code" , I should always be on the safe side. I have pinned to python 3.11. My experience with "mamba" is completely fine, works reliably.

What means `right Python interpreter`? Do you expect a certain revision, or special feature, or something more? Opening an editor-file that can do probably all python-versions. What kind of information derive you from an open file? Why can the python-version not be part of the dependency-list of ocp-cad-viewer and Build123d?

**Reply by bernhard-42** [reply]
nope, I just I just mean the one that the user wants to use.
OCP has requirements for Python but not the viewer.
